trieste/space.py

```diff
@@ -920,17 +917,242 @@ def __init__(self, spaces: Sequence[SearchSpace], tags: Optional[Sequence[str]]
             )
 
         self._spaces = dict(zip(tags, spaces))
+        self._tags = tuple(tags)  # avoid accidental modification by users
+
+    def __repr__(self) -> str:
+        """"""
+        return f"""CollectionSearchSpace(spaces =
```

Comment:
Maybe use a default implementation to avoid having to override?

```suggestion
        return f"""{self.__class__.__name__}(spaces =
```